internal/config/autosave.go

```diff
@@ -23,14 +23,20 @@ func SetAutoTime(a int) {
 
 func StartAutoSave() {
 	go func() {
+		autolock.Lock()
+		a := autotime
+		autolock.Unlock()
+		if a < 1 {
+			return
+		}
```

Comment:
Hmm, am I reading the code (not just your code) correctly: if `autosave` is already set to a positive value and we change it to another positive value, we spawn a new goroutine without stopping the old one? So both goroutines are now running? So autosave is actually done twice more often than it should? (And if we repeat that several more times, autosave will be done several times more often than it should, etc?)

So, all this looks broken, and your code seems to be a hack around this broken code, which fixes only a part of its broken behavior.

It seems the proper fix should be: let the goroutine run all the time, and just notify it about changes of the `autotime` value. The goroutine could use `select` to wait on both the timeout and the `autotime` change notifications.